gazebo/transport/Publisher.cc

```diff
@@ -268,6 +308,9 @@ void Publisher::Fini()
     TopicManager::Instance()->Unadvertise(this->topic, this->id);
 
   this->node.reset();
+
+  std::lock_guard<std::mutex> lock(pubMapMutex);
+  publisherMap.erase(this->id);
```

Comment:
I remember having a reason to destroy all publishers before the node, but I can't remember exactly why. Looking through the code, this pattern seems common:

```
this->pub.reset();
this->node->Fini();
this->node.reset();
```

If you don't have a strong reason for terminating the publisher after the node, I'd recommend switching order here.

Reply:
moved erase before resetting node c1e40a5d5a7ad2161c4b1aabbbf57d8430cba2e7